dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java

```diff
@@ -1538,4 +1650,342 @@ public void removeMemberUnprocessableTest() throws Exception {
             }
         }
     }
+
+    @Test
+    public void deleteGroupTest() throws Exception {
+
+        GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
+
+        Group parentGroup = null;
+
+        try {
+            context.turnOffAuthorisationSystem();
+
+            parentGroup = groupService.create(context);
+
+            context.commit();
+
+            parentGroup = context.reloadEntity(parentGroup);
```

Comment:
This method is missing a context.restoreAuthSystemState();   Also, could we add a comment as to why the `context.commit()` and `reloadEntity` are required in this test?  Most tests don't actually need those.  Are they needed here because you are doing a delete? Or could they be removed? 

(You might want to reverify the commit & reloadEntity are required after switching to using the GroupBuilder. I don't think they should be required, but I could be wrong here.)